[IMP] l10n_th localization update as per 19.4 (#18359)
odoo/documentation: content/applications/finance/fiscal_localizations/thailand.rst

:icon:`fa-book` :guilabel:`Report`, and select :guilabel:`P.N.D. 3 & 53 - WHT Report (TH)`.

Both P.N.D. 3 and P.N.D. 53 reports can be exported in `CSV` format, which is compatible with
the **RD Prep** application. To do so, follow these steps:

#. Make sure to filter by the appropriate month.
#. On the report page, click the :icon:`fa-cog` :guilabel:`(gear)` icon and select :guilabel:`PND3
   - Export for RD Prep (CSV)` or :guilabel:`PND 53 - Export for RD Prep (CSV)`.

This generates the :file:`Tax Report PND 3.csv` and :file:`Tax Report PND 53.csv` files that list
all the vendor bill lines with the applicable withholding tax.

.. _localizations/thailand/reports/50tawi:

Withholding tax certificate (50 Tawi)
-------------------------------------

The **50 Tawi**, or *Withholding Tax Certificate*, is a certificate issued by a payer to a payee
as official proof that tax has been withheld from a payment for remittance to the Revenue
Department.

To generate a `PDF` file of a withholding tax certificate:

#. Go to :menuselection:`Accounting --> Vendors --> Payments` and select the payment subject to
   withholding tax.

> #. Click the :guilabel:`Print 50 Tawi` button.

Reviewer: <img width="1918" height="807" alt="Image" src="https://github.com/user-attachments/assets/4a061720-4847-48f0-9bc3-205b0c8e9a42" />
<img width="1908" height="557" alt="Image" src="https://github.com/user-attachments/assets/dec08f67-9f3c-4f18-a1ef-4497bc646bed" />

I could not reproduce these steps on a fresh TH DB. In the screenshots attached neither have a  `Print 50 Tawi` button, although the option to withhold taxes can be enabled. Doing so requires additional steps not listed here.

Is there something specific to set up to have this option available? It might be worth documenting if it is the case :)

Author: Hello!

If you are on the latest build on master version of runbot, I think the reason is because the payment you created does not have a withholding tax. (We wanted to keep the flow as standard as possible, and this to appear only when it actually applies)

To make the process a bit easier, 
-> Create a vendor bill, one line is enough
-> For the tax, you have to use the one that says WHT. (If it doesn't appear, it is due to fiscal position. There is a standard configured taxes that are withholding).
-> Post the vendor bill, register payment. You will see register payment already has withholding lines calculated.
-> After registering payment, when you go to it, you will be able to see it 

Attaching video here!
https://drive.google.com/file/d/1p1kyiPWwRK8ddnmbnffhoenjkJlftp_c/view?usp=sharing

For the registering of withholding tax, there is a WIP to convert it to standard (https://www.odoo.com/odoo/project/967/tasks/5438849) so I think that would have to be a standard doc required soon.

.. _localizations/thailand/contacts:

Contacts
========

> The Thai localization utilizes specific fields on contact forms to generate Revenue Department

Reviewer: ```suggestion
The Thai localization utilizes specific fields on :doc:`contact forms <../../essentials/contacts`  to generate Revenue Department
```
Just a suggestion

reports and files used for submission.

.. _localizations/thailand/company:

Branch number
-------------

You can specify a company's branch number directly in the **Contacts** app. Open the company's
contact form, and click the `+` (plus) icon next to the :guilabel:`Tax ID` field to add the
:guilabel:`Branch Code`:

- If the contact is identified as a branch, input the Branch number in the :guilabel:`Branch Code`
  field.
- If the contact is a Headquarters, leave the :guilabel:`Branch Code` field blank.

.. _localizations/thailand/accounting:

Accounting
==========

.. _localizations/thailand/tax_invoice:

Tax invoice
-----------

The tax invoice PDF report can be generated from Odoo through the **Invoicing** or **Accounting**
app. Odoo defaults to issuance of tax invoice. To print commercial invoices, click the
:icon:`fa-cog` :guilabel:`(gear)` icon, click :guilabel:`Print`, and then select

> :guilabel:`Commercial Invoice`.

Reviewer: It might be worth mentioning the steps have to be completed on an invoice. The fact that it is a report made me look in the tax report section for a "Commercial invoice" filter.

<img width="1908" height="1166" alt="Image" src="https://github.com/user-attachments/assets/961083ff-1cb1-459c-a373-b00f5e92060d" />

.. _localizations/thailand/promptpay:

PromptPay QR code on invoices
-----------------------------

The PromptPay QR code is a QR code that can be added to invoices to allow customers to pay their
bills using the PromptPay-supported bank mobile application. The QR code is generated based on the
invoice amount and one of the following merchant information:

- Ewallet ID
- Merchant Tax ID
- Mobile Number

Activate QR codes
~~~~~~~~~~~~~~~~~

Go to :menuselection:`Accounting --> Configuration --> Settings`. Under the

> :guilabel:`Customer Payments` section, activate the :guilabel:`QR Codes` feature.

Reviewer: ```suggestion
:guilabel:`Customer Payments` section, activate the :guilabel:`Add QR-code link on PDF` feature.
```
It seems the precise name is this one. For some reason, SEPA is available on TH DB but SEPA is limited to EUR currency. This way, there's no confusion on what to activate

<img width="1169" height="311" alt="Image" src="https://github.com/user-attachments/assets/0486f6e4-68cb-4942-b934-33ea63c1d962" />

   * - :guilabel:`Thailand - Accounting Reports`
     - `l10n_th_reports`
     - This module includes the accounting reports for Thailand.

.. _localizations/thailand/specifics:

Localization overview
=====================

The Thai localization package ensures compliance with Thailand's fiscal and accounting regulations.
It includes tools for managing taxes, withholding taxes, reporting, and a predefined chart of
accounts tailored to Thai standards.

.. _localizations/thailand/taxes:

Taxes
-----

Installing the :guilabel:`Thailand - Accounting` (`l10n_th`) and :guilabel:`Thailand - Accounting
Reports` (`l10n_th_reports`) modules automatically creates default :doc:`taxes
<../accounting/taxes>` for :ref:`tax reports <localizations/thailand/tax_reports>` generation.
The package includes the following taxes:

- **VAT 7%**: For goods where VAT is recognized upon invoicing.
- **VAT 7% S**: For services where VAT is recognized upon payment.
- **VAT 0%**: For taxable goods or services charged at a zero rate, usually exports.
- **VAT-exempted**: For specific goods or services completely exempted from collection or payment
  of VAT.

> - **Withholding tax**: For income subject to payer withhold a portion of tax from payments.

Reviewer 2: I don't understand this description, could we change it? and maybe use this page for reference?https://www.odoo.com/documentation/master/applications/finance/accounting/taxes/retention.html

Author: Sure, that should work. 

On the side topic, recently, the module `Withholding on payment` was upgraded to `Withholding  Tax` and it covers more use cases where Thailand is one of the countries using the module.

Hmm, not sure how that will be documented because not all countries use that advanced `Withholding Tax` module and existing way of just using - tax still works for some countries.

For now, let me update it to reference that page

.. _localizations/thailand/income_tax_type:

Income tax type (Under section 40)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Income tax types are essential for generating **Withholding tax certificates (50 Tawi)** and
**P.N.D. 3 & 53** reports. These types ensure compliance with regulatory requirements.

> To configure this, go to :menuselection:`Accounting --> Configuration --> Taxes`. For taxes with

Reviewer 2: ```suggestion
Income tax types are essential for generating **Withholding tax certificates (50 Tawi)** and
**P.N.D. 3 & 53** reports and ensure compliance with regulatory requirements.

To configure income tax types, go to :menuselection:`Accounting --> Configuration --> Taxes`. For taxes with
```
(unless you meant something else by *this*? in any case I think we should clarify)

the :guilabel:`Purchase` tax type and :guilabel:`Withhold On Payment` enabled, select the
appropriate :guilabel:`Income Tax Type`.

.. _localizations/thailand/tax_reports:

Tax reports
-----------

Odoo generates the relevant VAT and WHT reports and allows users to export them as `CSV`
files compatible with RD Prep(:dfn:`Thai Revenue Department's official software used to format tax

> data for e-Filing.`).

Reviewer 2: ```suggestion
The relevant VAT and WHT reports can be generated and exported as `CSV`
files compatible with RD Prep(:dfn:`Thai Revenue Department's official software used to format tax
data for e-Filing.`).
```

.. _localizations/thailand/reports/pp30:

P.P. 30 - VAT report
~~~~~~~~~~~~~~~~~~~~

The **P.P. 30** is a monthly value-added tax (VAT) return report required by the Revenue
Department of Thailand. It summarizes the output and input VAT for the period.

The P.P. 30 report can be exported in `CSV` format, compatible with the **RD Prep**. To do so,
follow these steps:

> #. To access it, navigate to :menuselection:`Accounting --> Reporting --> Tax Report`, click

Reviewer 2: ```suggestion
#. Navigate to :menuselection:`Accounting --> Reporting --> Tax Report`, click
```

   :icon:`fa-book` :guilabel:`Report`, and select :guilabel:`P.P.30 - VAT Report`.
#. Make sure to filter by the appropriate date or date range.
#. On the report page, click the :icon:`fa-cog` :guilabel:`(gear)` icon and select
   :guilabel:`Export for RD Prep (CSV)`.

.. _localizations/thailand/reports/sales_purchase:

Sales and purchase tax reports
******************************

> The **Sales and Purchase Tax Reports** detail all input and output VAT transactions for a specific

Reviewer 2: nitpick but I don't think the bold and capital letters are needed here

period, serving as a mandatory compliance document that VAT-registered businesses must maintain for
record-keeping and present to the Revenue Department during tax audits.

> The Sales and Purchase Tax Reports can be exported as `XLSX` files. To do so, follow these steps:

Reviewer 2: ```suggestion
The sales and purchase tax reports can be exported as `XLSX` files. To do so, follow these steps:
```

#. Make sure to filter by the appropriate month.
#. Click the :icon:`fa-cog` :guilabel:`(gear)` icon and select :guilabel:`Sales Tax Report (xlsx)`
   or :guilabel:`Purchase Tax Report (xlsx)`.

.. _localizations/thailand/reports/pnd:

P.N.D. 3 & 53 - WHT report
~~~~~~~~~~~~~~~~~~~~~~~~~~

The **P.N.D. 3** and **P.N.D. 53** reports list all the transactions within the period subject to
the withholding of income tax from vendor bills. This applies when a company makes payments for
services received (such as rental, hiring, transportation, insurance, management fees, or
consulting) and is required to withhold "Personal (PND3)" or "Corporate (PND53)" income tax.

> To access it, navigate to :menuselection:`Accounting --> Reporting --> Tax Report`, click

Reviewer 2: ```suggestion
To access them, navigate to :menuselection:`Accounting --> Reporting --> Tax Report`, click
```
(as there are 2 reports as far as I understand? even if they seem to be grouped together in the report view?)

   withholding tax.
#. Click the :guilabel:`Print 50 Tawi` button.

.. _localizations/thailand/contacts:

Contacts
========

The Thai localization utilizes specific fields on
:doc:`contact forms <../../../../essentials/contacts` to generate Revenue Department reports and
files used for submission.

.. _localizations/thailand/company:

Branch number
-------------

You can specify a company's branch number directly in the **Contacts** app. Open the company's
contact form, and click the :icon:`fa-plus` :guilabel:`(plus)` icon next to the :guilabel:`Tax ID`
field to add the :guilabel:`Branch Code`:

- If the contact is identified as a branch, input the Branch number in the :guilabel:`Branch Code`
  field.
- If the contact is a Headquarters, leave the :guilabel:`Branch Code` field blank.

.. note::
   Only companies set in :guilabel:`Thailand` feature the :guilabel:`Branch Code` field.

.. _localizations/thailand/accounting:

Accounting
==========

.. _localizations/thailand/tax_invoice:

Tax invoice
-----------

The tax invoice PDF can be generated from Odoo through the **Invoicing** or **Accounting**

> app. To print a tax invoice, follow the steps to :ref:`create an invoice

Reviewer 2: Unless there's a specific reason you wanted to mention the two apps, I would remove the first sentence

> <accounting/invoice/creation>` and click :guilabel:`Print` button to download the `PDF` tax

Reviewer 2: ```suggestion
<accounting/invoice/creation>` and click :guilabel:`Print` to download the PDF tax
```

invoice.

To print commercial invoices, click the :icon:`fa-cog` :guilabel:`(gear)` icon, click
:guilabel:`Print`, and then select :guilabel:`Commercial Invoice`.

.. _localizations/thailand/promptpay:

PromptPay QR code on invoices
-----------------------------

> The PromptPay QR code is a QR code that can be added to invoices to allow customers to pay their

Reviewer 2: ```suggestion
The PromptPay QR code can be added to invoices to allow customers to pay their
```

bills using the PromptPay-supported bank mobile application. The QR code is generated based on the
invoice amount and one of the following merchant information:

- Ewallet ID
- Merchant Tax ID
- Mobile Number

.. _localizations/thailand/activateqr:

Activate QR codes on invoice
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Go to :menuselection:`Accounting --> Configuration --> Settings`. Under the :guilabel:`Invoice

> Online Payment` option, check the :guilabel:`Add QR-code Link on PDF` feature.

Reviewer 2: given how small this is, I think I wouldn't have a separate section for this and instead have the instructions directly under the previous paragraph and I'd change it as follows:
```suggestion
To activate PromptPay QR codes on invoices, go to :menuselection:`Accounting --> Configuration --> Settings` and enable the :guilabel:`Add QR-code Link on PDF` feature under the :guilabel:`Invoice Online Payment` option.
```
or something similar

.. _localizations/thailand/configureqr:

> Configure QR code on bank account

Reviewer 2: ```suggestion
Bank account configuration
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Go to :menuselection:`Accounting --> Configuration --> Journals`, and open the journal linked to the

> bank account requiring the use of PromptPay QR codes.

Reviewer 2: ```suggestion
To configure the bank account for PromptPay QR codes, go to :menuselection:`Accounting --> Configuration --> Journals`, and open the journal linked to the
relevant bank account.
```

Under :guilabel:`Journal Entries` tab, fill in the bank account number:

- For an existing account: Hover over the account number and click the :icon:`oi-arrow-right` icon.
- For a new account: Type the account number and click :guilabel:`Create`.

In the bank account form view, locate the :guilabel:`EMV QR Settings` tab. Select your

> :guilabel:`Proxy Type` and enter the corresponding :guilabel:`Proxy Value`.

Reviewer 2: ```suggestion
In the bank account form view, go to the :guilabel:`EMV QR Settings` tab. Select the relevant
:guilabel:`Proxy Type` and enter the corresponding :guilabel:`Proxy Value`.
```
I didn't test this, so I'm wondering if it's really obvious to the user which proxy type and proxy value they need to select?

Author: yes indeed, proxy type is something like `mobile number` or `bank account number` so they will know what to fill in. 

On my end, I intend to create a task to improve the UI of the following view to make it easier.

>    :alt: PromptPay bank account configuration

Reviewer: <img width="1912" height="416" alt="Image" src="https://github.com/user-attachments/assets/072d2085-a307-45ea-b6ca-097f5f9b1d75" />

```suggestion
Open a contact's form from :menuselection:`Customers --> Customers` or :menuselection:`Vendors --> Vendors`, click the :guilabel:`Accounting` tab, and select a **bank account** for which you want to activate PromptPay QR.  Set the :guilabel:`Proxy Type` and fill in the [...]
```
Please note that the Contacts app isn't installed along with the Accounting app, and it might therefore be confusing for users only having Accounting or Invoicing to look for Contacts. I've made a suggestion on how they can get to a contact's form without having the contacts app.

Also, I couldn't find `Bank Accounts` under `Configuration` (see screenshot). I had to go to the `Accounting` tab of a contact.

<img width="1254" height="810" alt="Image" src="https://github.com/user-attachments/assets/6dfbc66f-6ce4-4b45-a8e2-d7c370b84c1b" />

Author: Thank you, I did not think to update this QR page, it seems it has changed a bit since the past version.
Let me update it accordingly base on master and push again
